### PR TITLE
Typing Updates

### PR DESCRIPTION
**Pull request by lishaduck**
This doesn't touch implementations, thus the branch name.

Depends on #165

Here's the diff as GH doesn't play well with stacks from forks: <https://github.com/lishaduck/node-elm-review/compare/jsdocs...pure-typings>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
     "prettier-check": "prettier . --check",
     "prettier-fix": "prettier . --write",
     "tsc": "tsc && tsc --project tsconfig.no-implicit-any.json",
-    "tsc-watch": "tsc --watch",
```
Comment (jfmengels) [inline package.json:18]: How can we run `tsc` in watch mode using `turbo`? :thinking:

Comment (lishaduck) [reply package.json:18]: `turbo watch tsc`
It'll cache a bit more, and isn't any more characters.

**Comment by lishaduck** [comment]
Ah, silly me. I pushed the fix to my next branch. I'll cherry-pick it soonish, and this should be good to go.

**Comment by jfmengels** [comment]
Alright.
Note that I just merged a PR by someone else (https://github.com/jfmengels/node-elm-review/pull/162, the final commit in `main` being 7ba6bd7ca1af522d21bebe89a66c44ed7d79f211), which may conflict with this branch. If I'm not mistaken, it's only the rename from `ruleType` to `_ruleType` that need to be skipped.

File: package.json (hunk 1 above)
Comment (jfmengels) [reply package.json:17]: I'm getting this error when trying it:
```
• Packages in scope: 
• Running tsc in 0 packages
• Remote caching disabled
  × failed to connect to daemon
  ╰─▶ server is unavailable: channel closed
```

**Comment by lishaduck** [comment]
> Alright. Note that I just merged a PR by someone else (#162, the final commit in `main` being [7ba6bd7](https://github.com/jfmengels/node-elm-review/commit/7ba6bd7ca1af522d21bebe89a66c44ed7d79f211)), which may conflict with this branch. If I'm not mistaken, it's only the rename from `ruleType` to `_ruleType` that need to be skipped.

Thanks for letting me know!

**Comment by jfmengels** [comment]
CI is failing:
> Run turbo run test
> • Running test
> • Remote caching disabled
>   x root task //#test (turbo testing) looks like it invokes turbo and might
>   | cause a loop

I think I should have updated CI to run either `npm test` or `turbo testing` instead of `turbo testing`. Does that sound correct to you?

**Comment by lishaduck** [comment]
> CI is failing:
> 
> > Run turbo run test
> > • Running test
> > • Remote caching disabled
> > x root task //#test (turbo testing) looks like it invokes turbo and might
> > | cause a loop
> 
> I think I should have updated CI to run either `npm test` or `turbo testing` instead of `turbo testing`. Does that sound correct to you?

Yeah. I fixed it :)

**Comment by lishaduck** [comment]
> Yeah. I fixed it :)

I take that back. What's up with haste-map? It looks like jest is failing now.

File: package.json (hunk 1 above)
Comment (lishaduck) [reply package.json:18]: Hmm. Ok, I'll revert it.

**Comment by jfmengels** [comment]
Could it be that `turbo` / CI is caching results from previous runs and that that causes problems?

**Comment by lishaduck** [comment]
> Could it be that `turbo` / CI is caching results from previous runs and that that causes problems?

I don't think so:
> cache bypass, force executing 2d4e737178c4d703
It's not using a cache. Does jest need to run before test-run? I did parallelize it, which might be the issue.

**Comment by lishaduck** [comment]
> > Could it be that `turbo` / CI is caching results from previous runs and that that causes problems?
> 
> I don't think so:
> 
> > cache bypass, force executing 2d4e737178c4d703
> 
> So it's not using the cache. Does jest need to run before test-run? I did parallelize it, which might be the issue.

Hmmm. Locally, test-run works in parallel, which is good. I must've just corrupted it by quitting npm at a bad time. jest is crashing in elm-tooling.test. I'll look into it. After all, this shouldn't change impls. That's the point of this branch.

**Comment by lishaduck** [comment]
Oh, could it be --detectOpenHandles? I added that to catch some regressions I made, maybe there were some other ones?

> > Could it be that `turbo` / CI is caching results from previous runs and that that causes problems?
> 
> I don't think so:
> 
> > cache bypass, force executing 2d4e737178c4d703
> 
> So it's not using the cache. Does jest need to run before test-run? I did parallelize it, which might be the issue.

Also, it's reproducible with just `npm run jest`.
Does jest redirect output? console.log at the top of elm-tooling.test.js never logs.

**Comment by lishaduck** [comment]
I figured it out. Somehow, elm-review is parsing jest flags as it's own, but then the log gets overriden by jest so you can't read it. 
It's probably a bug, but it can go into #156 for now.

**Comment by jfmengels** [comment]
Thank you for looking into all of this! :pray:

**Pull request merged**